### DISCUSSION TITLE
PnP dependencies priority

### DISCUSSION THREAD

**Discussion by justsaul** [yarnpkg/berry#5395] [Q&A] [answered]
It appears that `peerDependencies` have `higher priority` over regular dependencies when consuming packages.

Meaning if we have such set-up:

```JSON
// test-app package.json
{
  "name": "test-app",
  "private": true,
  "dependencies": {
    "test-package": "20.0.0",
    "react": "17.0.0-rc.3"
  }
}

// test-package package.json
{
  "name": "test-package",
  "version": "20.0.0",
  "private": true,
  "dependencies": {
    "react": "18.2.0"
  },
  "peerDependencies": {
    "react": "*"
  },
}
```

when `test-package` functionality will be used within `test-app`, `test-package` will use react that comes from `test-app` (17.0.0-rc.3)  instead of using it's internal react version that comes from its dependencies (18.2.0).

This behaviour makes total sense, however, since I couldn't find any documentation around this, I wanted to ask if it's intended and stable behaviour?

**Comment by merceyz** [comment, marked as answer]
Yes, it's documented here https://yarnpkg.com/configuration/manifest#peerDependencies

> Note that peer dependencies can also be listed as regular dependencies; in this case, Yarn will use the package provided by the ancestors if possible, but will fallback to the regular dependencies otherwise
